resolwe/storage/tests/test_listener.py

```diff
@@ -0,0 +1,84 @@
+# pylint: disable=missing-docstring
+from unittest.mock import MagicMock, patch
+
+from resolwe.flow.managers.listener import ExecutorListener
+from resolwe.flow.managers.protocol import ExecutorProtocol
+from resolwe.storage.models import FileStorage, ReferencedPath, StorageLocation
+from resolwe.test import TestCase
+
+
+class ListenerTest(TestCase):
+    def setUp(self):
+        self.listener = ExecutorListener()
+        self.file_storage = FileStorage()
+        self.file_storage.save()
+        self.storage_location = StorageLocation(
+            file_storage=self.file_storage, connector_name="GCS", status="OK"
+        )
+        self.storage_location.save()
+        path = ReferencedPath.objects.create(
+            path="test.me", md5="md5", crc32c="crc", awss3etag="aws"
+        )
+        self.storage_location.files.add(path)
+
+    def test_download_finished(self):
+        storage_location = StorageLocation(
+            file_storage=self.file_storage, connector_name="local"
+        )
+        obj = {
+            "command": ExecutorProtocol.DOWNLOAD_FINISHED,
+            "data_id": -1,
+            "storage_location_id": -2,
+        }
+        send_wrapper = MagicMock()
+        send_reply = MagicMock()
+        async_to_sync = MagicMock(return_value=send_wrapper)
+        with patch(
```

Comment:
It would be more readable if you patch the whole function instead of using `with` block.

```
@patch(...)
def test(self, mock):
   ...
```